sdk/src/wallet/operations/transaction/high_level/account_block_issuer_keys.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use alloc::collections::BTreeSet;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::{
+        api::{
+            transaction_builder::{transition::AccountChange, Transitions},
+            PreparedTransactionData,
+        },
+        secret::SecretManage,
+        ClientError,
+    },
+    types::block::output::{feature::BlockIssuerKey, AccountId},
+    wallet::{operations::transaction::TransactionOptions, types::TransactionWithMetadata, Wallet, WalletError},
+};
+
+/// Params `modify_account_block_issuer_key()`
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ModifyAccountBlockIssuerKey {
+    pub account: AccountId,
+    /// The keys that will be added.
+    pub keys_to_add: Vec<BlockIssuerKey>,
+    /// The keys that will be removed.
+    pub keys_to_remove: Vec<BlockIssuerKey>,
+}
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    WalletError: From<S::Error>,
+    ClientError: From<S::Error>,
+{
+    pub async fn modify_account_output_block_issuer_keys(
+        &self,
+        params: ModifyAccountBlockIssuerKey,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> Result<TransactionWithMetadata, WalletError> {
+        let options = options.into();
+        let prepared_transaction = self
+            .prepare_modify_account_output_block_issuer_keys(params, options.clone())
+            .await?;
+
+        self.sign_and_submit_transaction(prepared_transaction, options).await
+    }
+
+    /// Prepares the transaction for [Wallet::create_account_output()].
+    pub async fn prepare_modify_account_output_block_issuer_keys(
+        &self,
+        params: ModifyAccountBlockIssuerKey,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> Result<PreparedTransactionData, WalletError> {
+        log::debug!("[TRANSACTION] prepare_modify_account_output_block_issuer_keys");
+
+        let change = AccountChange::ModifyBlockIssuerKeys {
+            keys_to_add: params.keys_to_add,
+            keys_to_remove: params.keys_to_remove,
+        };
+
+        let account_id = params.account;
+
+        let mut options = options.into();
+        if let Some(options) = options.as_mut() {
+            if let Some(transitions) = options.transitions.take() {
+                options.transitions = Some(transitions.add_account(account_id, change));
+            }
+        } else {
+            options.replace(TransactionOptions {
+                transitions: Some(Transitions::new().add_account(account_id, change)),
+                required_inputs: BTreeSet::from([self
+                    .get_account_output(account_id)
+                    .await
+                    .ok_or(WalletError::AccountNotFound)?
+                    .1
+                    .output_id]),
```

Comment:
Could you also add this when the options are passed in?

Reply:
Alternatively, requirements for transitions could be added in the transaction builder based on the passed in `Transitions`, which make make more sense

```rust
/// Account requirement.
Account(AccountId),
```

Reply:
@Thoralf-M modified this part and I'm not sure what you mean, so I'm afraid I can't make those changes. If I could push to this repo directly, then I guess you could directly push those changes to that branch, but I would need to get the permission.

Reply:
@DaughterOfMars you mean something like this?
```Rust
    pub fn finish(mut self) -> Result<PreparedTransactionData, TransactionBuilderError> {
        if let Some(transitions) = &self.transitions {
            for (account_id, _) in transitions.accounts() {
                self.requirements.push(Requirement::Account(*account_id))
            }
        }
```

Reply:
Yeah, exactly

Reply:
Should be good now

Reply:
For begin staking not, forgot to remove it there

Reply:
Removed